### ISSUE TITLE
Documentation: docs should be generated for "internal" group 

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1292]
Documentation is not generated for items in the "internal" group. Example:

https://github.com/igraph/igraph/blob/master/include/igraph_datatype.h#L34

Does anyone remember why this is so?

Is there currently a *simple* way to include this group?

I propose we include it in the standard documentation. The igraph project does not have enough manpower, so we should make the barrier to contributions as small as possible.

**Comment by ntamas** [comment]
Well, it is not included in the documentation because, as its name implies, its an internal thing and it should not concern the user :) In many cases the internal data types are not even documented properly, so I think including this section in the documentation now would probably do more harm than good. We could aim for documenting internal stuff properly (to an extent that matches the documentation quality of the public part of the library), but that's probably for 0.9.

**Comment by vtraag** [comment]
Yes, it would be good to have proper documentation of the internal types and functions. But I agree with @ntamas that it probably wouldn't be the right time to do that now. But it would definitely help new contributors to find their way around the internals of `igraph`, so I think this is quite important for making it as easy as possible to get new contributions in.
